backend/chatbot/chat_router.py

```
@@ -25,209 +32,143 @@
-                    detail="Invalid or expired token"
-                )
+            if response.status_code != 200:
+                raise HTTPException(status_code=401, detail="Invalid or expired token")
             return response.json()
     except httpx.TimeoutException:
-        logger.error("Token verification timeout")
-        raise HTTPException(
-            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
-            detail="Authentication service unavailable"
-        )
+        raise HTTPException(status_code=503, detail="Authentication service unavailable")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)